### PR TITLE
[python] Neaten registration-mapping test code

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** Issue #2242

**Changes:** This is some factoring to help simplify #2281, which will be rebased on top of this PR.

**Notes for Reviewer:**

### CODE REVIEW

File: apis/python/tests/test_registration_mappings.py [hunk 1]
```diff
@@ -399,17 +407,29 @@ def test_isolated_soma_experiment_mappings(soma1):
...
     solo_experiment_first,
 ):
+    h5ad1 = create_h5ad_canned(1, obs_field_name, var_field_name)
+    h5ad2 = create_h5ad_canned(2, obs_field_name, var_field_name)
+    h5ad3 = create_h5ad_canned(3, obs_field_name, var_field_name)
+    h5ad4 = create_h5ad_canned(4, obs_field_name, var_field_name)
+
+    import shutil
+
+    shutil.copy(h5ad1, "/tmp/1.h5ad")
```
Comment (ryan-williams) [inline apis/python/tests/test_registration_mappings.py:428]: should this be a `TemporaryDirectory`, instead of hard-coded `/tmp`?

**Review by ryan-williams — APPROVED** [review]
lgtm, one q though, can be addressed as a follow-on if necessary

File: apis/python/tests/test_registration_mappings.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/tests/test_registration_mappings.py:428]: Derp, those were debugs to take back out. Thanks!

**Comment by johnkerl** [comment]
https://github.com/single-cell-data/TileDB-SOMA/pull/2285/commits/e605dbc1fd6f49db80586bb8aacb14b0321e9995 -- thank you @ryan-williams !

**Pull request merged**